core-model/src/main/java/com/sequenceiq/cloudbreak/domain/stack/cluster/Cluster.java

```diff
@@ -170,6 +170,9 @@
     @OneToOne(mappedBy = "cluster", cascade = {CascadeType.PERSIST, CascadeType.REMOVE}, orphanRemoval = true)
     private Gateway gateway;
 
+    @OneToOne(mappedBy = "cluster", cascade = {CascadeType.PERSIST, CascadeType.REMOVE}, orphanRemoval = true)
+    private IdBroker idBroker;
```

Comment:
do we really need this side of the relation? This would mean every time we fetch `Cluster` from db, it would fetch `IdBroker` also. If possible this should be omitted or converted to lazy

Reply:
Yea theres some parts where I think i need to get Idbroker from the cluster to pass it into the GeneralClusterConfigs.  By default it's set to lazy loading.

Reply:
OneToOne is eager by default and I think @lacikaaa is worried about the performance as all the stack queries would fetch this entity as well while we need it once when we pass it to the pillar.

Reply:
Ah right, i was thinking about ManyToOne/many. I'll fix.

Reply:
I would just drop it from the `Cluster` entity and fetch it with a separate repository call when it needed. This way you can avoid lazy exceptions.